### PR TITLE
Data Distribution Service requires improved input validity checking

### PR DESCRIPTION
**Pull request by skolli853780**
1) Added input id validation
2) Add checks for No records found, bad requests, internal server errors etc..

### CODE REVIEW

File: src/main/java/gov/nist/oar/ds/controller/DownloadController.java [hunk 1]
```diff
@@ -144,7 +145,7 @@ public void setDownloadService(DownloadService downloadService) {
    */
   @RequestMapping(value = "/{dsId}", method = RequestMethod.GET)
 
-  public ResponseEntity<byte[]> downloadZipFile(String id) throws Exception {
+  public ResponseEntity<byte[]> downloadZipFile(@RequestParam(value = "id", required = true) String id) throws Exception {
```
Comment (RayPlante) [inline src/main/java/gov/nist/oar/ds/controller/DownloadController.java]: Som -- what happens if an id parameter is not present?

Comment (skolli853780) [reply src/main/java/gov/nist/oar/ds/controller/DownloadController.java]: @RayPlante  As per my email, Deoyani's PR should take care of all the exception handling scenarios for zip and single file download.. Once Deoyani's PR is merged, we can close this PR without merging into integration.

Comment (RayPlante) [reply src/main/java/gov/nist/oar/ds/controller/DownloadController.java]: concur.

**Comment by RayPlante** [comment]
Changes in #39 cover fixes offered here (see above comments).

**Pull request closed without merging**